source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -260,13 +260,10 @@ namespace aspect
             velocity_gradient_3d[2][2] = velocity_gradient[2][2];
           }
 
-        std::vector<unsigned int> deformation_types;
-        std::vector<double> volume_fraction_mineral;
-
         for (unsigned int mineral_i = 0; mineral_i < n_minerals; ++mineral_i)
           {
 
-            ref_deformation_type(data_position,data,mineral_i) = (unsigned int)DeformationType::passive;
+            set_deformation_type(data_position,data,mineral_i,(unsigned int)DeformationType::passive);
```

Comment:
We try to avoid C-style casts, better `static_cast<unsigned int>(DeformationType::passive)`